re2: fix for pkgsStatic

tobim opened the pull request:
Not using `lib.optional` to avoid changing the `.drv` hash.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

tobim committed:
re2: fix for pkgsStatic

SuperSandro2000 commented on pkgs/development/libraries/re2/default.nix:
```diff
@@ -17,10 +17,14 @@ stdenv.mkDerivation {
     substituteInPlace Makefile  --replace "SED_INPLACE=sed -i '''" "SED_INPLACE=sed -i"
   '';
 
+  buildFlags = if stdenv.hostPlatform.isStatic then [ "static" ] else null;
```
```suggestion
  buildFlags = lib.optionals stdenv.hostPlatform.isStatic [ "static" ];;
```

SuperSandro2000 commented on pkgs/development/libraries/re2/default.nix:
```diff
@@ -17,10 +17,14 @@ stdenv.mkDerivation {
   doCheck = true;
   checkTarget = "test";
 
+  installTargets = if stdenv.hostPlatform.isStatic then [ "static-install" ] else null;
```
```suggestion
  installTargets = lib.optionals stdenv.hostPlatform.isStatic [ "static-install" ];
```

SuperSandro2000 reviewed (changes requested):
> Not using `lib.optional` to avoid changing the `.drv` hash.

This is causing 26 rebuilds. Please do it properly.

tobim commented:
> This is causing 26 rebuilds.

ofBorg disagrees.

SuperSandro2000 commented:
> ofBorg disagrees.

Using lib.optional does 26 rebuilds which is totally fine to merge into master. Please only use rebuild avoiding hacks when it would cause stdenv rebuilds. This is not such change so we don't need to do it.

tobim committed:
Don't use rebuild-avoiding hacks

`lib.optionals` makes the code more readable. A few rebuilds because of that are tolerable.

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

tobim commented:
@SuperSandro2000 mind having another look?

SuperSandro2000 commented:
Checked it manually and it builds.
